### PR TITLE
Add  testcases with ` Image` , covering all methods.

### PR DESCRIPTION
**Pull request by chchliang**
Signed-off-by: chchliang <chen.chuanliang@zte.com.cn>

Add  testcases with ` Image` , covering all methods.

### CODE REVIEW

**Comment by chchliang** [comment]
@GordonTheTurtle 
Here haven't modified.

07:27:04 ----------------------------------------------------------------------
07:27:04 FAIL: docker_api_swarm_test.go:349: DockerSwarmSuite.TestAPISwarmRaftQuorum
07:27:04 
07:27:04 [d7e43d3b77374] waiting for daemon to start
07:27:04 [d7e43d3b77374] daemon started
07:27:04 
07:27:04 [de365303bf825] waiting for daemon to start
07:27:04 [de365303bf825] daemon started
07:27:04 
07:27:04 [dbd370546fecf] waiting for daemon to start
07:27:04 [dbd370546fecf] daemon started
07:27:04 
07:27:04 [de365303bf825] exiting daemon
07:27:04 [dbd370546fecf] exiting daemon
07:27:04 docker_api_swarm_test.go:374:
07:27:04     c.Assert(err.Error(), checker.Contains, "deadline exceeded")
07:27:04 ... obtained string = "Error response from daemon: rpc error: code = Unknown desc = raft: failed to process the request: node lost leader status"
07:27:04 ... substring string = "deadline exceeded"
07:27:04 
07:27:04 [d7e43d3b77374] exiting daemon
07:27:54 
07:27:54 ----------------------------------------------------------------------

**Comment by thaJeztah** [comment]
restarted CI

File: image/image_test.go [hunk 1]
```diff
@@ -54,6 +55,28 @@ func TestMarshalKeyOrder(t *testing.T) {
 	}
 }
 
+func TestImageWithAllMethods(t *testing.T) {
```
Comment (dnephin) [inline image/image_test.go]: nit: maybe just `TestImage()`

**Review by dnephin — APPROVED** [review]
LGTM

File: image/image_test.go [hunk 2]
```diff
@@ -54,6 +55,28 @@ func TestMarshalKeyOrder(t *testing.T) {
...
+	var cid = "50a16564e727"
+	var raw = []byte("279533a6d3a9")
```
Comment (thaJeztah) [inline image/image_test.go]: perhaps this should be a more "JSON" value; also; this is already tested in `TestNewFromJSON()` above, so I don't think we need to test that again here

File: image/image_test.go [hunk 3]
```diff
@@ -54,6 +55,28 @@ func TestMarshalKeyOrder(t *testing.T) {
...
+			Config: config,
+		},
+		rawJSON:    raw,
+		computedID: ID(cid),
+	}
+
+	assert.Equal(t, cid, img.ImageID())
+	assert.Equal(t, cid, img.ID().String())
+	assert.Equal(t, raw, img.RawJSON())
+	assert.Equal(t, platform, img.Platform())
```
Comment (thaJeztah) [inline image/image_test.go]: This tests only the case where `img.OS` is empty, but not if it _is_ set. Perhaps this should in a separate test and it should test both if `img.OS` is set, and `img.OS` is empty (see https://github.com/moby/moby/blob/1daa10301acfe0742dd852dfb8603c48afdaa1ee/image/image.go#L99-L106)

File: image/image_test.go [hunk 4]
```diff
@@ -54,6 +55,28 @@ func TestMarshalKeyOrder(t *testing.T) {
...
+	assert.Equal(t, config, img.RunConfig())
+
```
Comment (thaJeztah) [inline image/image_test.go]: Remove this blank line

File: image/image_test.go [hunk 5]
```diff
@@ -54,6 +55,28 @@ func TestMarshalKeyOrder(t *testing.T) {
 	}
 }
 
+func TestImageWithAllMethods(t *testing.T) {
+	var cid = "50a16564e727"
+	var raw = []byte("279533a6d3a9")
+	config := &container.Config{}
```
Comment (thaJeztah) [inline image/image_test.go]: Maybe there should be _something_ in this config so that we're not comparing an empty config

**Review by thaJeztah — CHANGES_REQUESTED** [review]
left some notes

**Comment by chchliang** [comment]
@thaJeztah 
I haved modified and add `TestPlatformWithOSNotEmpty`  testcase.

**Comment by chchliang** [comment]
@thaJeztah 

07:48:05 PASS: docker_api_swarm_test.go:189: DockerSwarmSuite.TestAPISwarmPromoteDemote	5.977s
07:48:20 
07:48:20 ----------------------------------------------------------------------
07:48:20 FAIL: docker_api_swarm_test.go:349: DockerSwarmSuite.TestAPISwarmRaftQuorum
07:48:20 
07:48:20 [d6a375f5884b7] waiting for daemon to start
07:48:20 [d6a375f5884b7] daemon started
07:48:20 
07:48:20 [d328969bba510] waiting for daemon to start
07:48:20 [d328969bba510] daemon started
07:48:20 
07:48:20 [dcaae55f920a8] waiting for daemon to start
07:48:20 [dcaae55f920a8] daemon started
07:48:20 
07:48:20 [d328969bba510] exiting daemon
07:48:20 [dcaae55f920a8] exiting daemon
07:48:20 docker_api_swarm_test.go:374:
07:48:20     c.Assert(err.Error(), checker.Contains, "deadline exceeded")
07:48:20 ... obtained string = "Error response from daemon: rpc error: code = Internal desc = transport is closing"
07:48:20 ... substring string = "deadline exceeded"
07:48:20 
07:48:20 [d6a375f5884b7] exiting daemon
07:49:11 
07:49:11 ----------------------------------------------------------------------

File: image/image_test.go (hunk 1 above)
Comment (thaJeztah) [reply image/image_test.go]: Yes, I think just `TestImage()` would be better

File: image/image_test.go [hunk 6]
```diff
@@ -54,6 +55,39 @@ func TestMarshalKeyOrder(t *testing.T) {
...
+		computedID: ID(cid),
+	}
+
+	assert.Equal(t, cid, img.ImageID())
+	assert.Equal(t, cid, img.ID().String())
+	assert.Equal(t, platform, img.Platform())
+	assert.Equal(t, config, img.RunConfig())
+}
+
+func TestImageWithPlatformNotEmpty(t *testing.T) {
```
Comment (thaJeztah) [inline image/image_test.go]: This could also just be `TestImagePlatformNotEmpty()`

**Review by thaJeztah — CHANGES_REQUESTED** [review]
thanks @chchliang - two minor nits, but LGTM after that . If you update, that would also re-trigger the (non related) CI failure; that test is a bit flaky it looks like 😅

**Comment by chchliang** [comment]
Hi @thaJeztah ,i haved modified, all are green.

**Review by thaJeztah — APPROVED** [review]
thank you! LGTM

**Pull request merged**